### PR TITLE
Add github actions CI inspired by travis/appveyor scripts

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This PR adds a few CI builds on Github actions that I have put together form a CI file of my own and taking inspiration from the existing travis/appveyor CI scripts.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            CXX: icpc
+            INSTALL_ONEAPI: true
+          - name: build-ubuntu-icpx
+            CXX: icpx
+            INSTALL_ONEAPI: true
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: install OneAPI
```
Comment (amadio) [inline .github/workflows/ci.yaml:45]: Maybe we can just use their docker image? https://github.com/intel/oneapi-containers

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            subset: sse
+          - name: build-windows-avx
+            subset: avx
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+    - name: ctest
+      run: |
+        ctest -V -S test.cmake
```
Comment (amadio) [inline .github/workflows/ci.yaml]: Missing line break.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+jobs:
+  #clang-format:
+  #  runs-on: ubuntu-latest
+  #  steps:
+  #  - uses: actions/checkout@v2
+  #  - uses: DoozyX/clang-format-lint-action@v0.12
+  #    with:
+  #      exclude: './thirdparty'
+  #      clangFormatVersion: 12
```
Comment (bernhardmgruber) [inline .github/workflows/ci.yaml:13]: When applying clang-format there is a huge diff. I guess we leave this for now. We also need to settle on a clang-format version and probably update the `.clang-format` file as well.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (bernhardmgruber) [reply .github/workflows/ci.yaml:45]: We could! I am not familiar with docker so I would need some help if we want to pursue this. However, what is wrong with just installing it via Intel's apt repository?

Comment (amadio) [reply .github/workflows/ci.yaml:45]: For GitHub Actions, it should boil down to just adding `image: intel/oneapi-basekit` at the right place in the file...

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (amadio) [reply .github/workflows/ci.yaml:13]: I'd leave this out for now, formatting changes are irrelevant if the project is in life-support mode.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (bernhardmgruber) [reply .github/workflows/ci.yaml:45]: I just tried using their docker image, but then the submodule checkout fails: https://github.com/bernhardmgruber/Vc/runs/2885172514?check_suite_focus=true
Also the container initialization adds 8min to the CI time.

**Comment by bernhardmgruber** [comment]
I think we need to decide which compilers we care about: https://github.com/bernhardmgruber/Vc/actions/runs/961463706
VS2019 and ipcx fail the tests because their results diverge from the reference results more ULPs than other compilers. We could fix that, remove those compilers from the CI or just flag those CI jobs as not mandatory for PRs.

**Comment by amadio** [comment]
We need to add `-fp-model=precise` for ICC and related compilers, I think. They are by default a bit too aggressive with unsafe arithmetic/math optimizations.

**Comment by bernhardmgruber** [comment]
> We need to add `-fp-model=precise` for ICC and related compilers, I think.

Good guess. Vc detects `icpc` and `icc` and sets `-fp-model source`. That could solve `icpx`.
The Visual Studio 2019 builds are still failing though.

**Comment by bernhardmgruber** [comment]
I removed the failing CI runs for now and will create a separate PR for them later.

**Review by agheata — APPROVED** [review]
Looks good!

**Pull request merged**